opendevin/controller/command_manager.py

```diff
@@ -0,0 +1,70 @@
+from typing import List
+
+from opendevin.observation import CmdOutputObservation
+from opendevin.sandbox.sandbox import DockerInteractive
+
+
+class BackgroundCommand:
+    def __init__(self, id: int, command: str, dir: str):
+        self.command = command
+        self.id = id
+        self.shell = DockerInteractive(id=str(id), workspace_dir=dir)
+        self.shell.execute_in_background(command)
+
+    def get_logs(self) -> str:
+        # TODO: get an exit code if process is exited
+        return self.shell.read_logs()
+
+
+class CommandManager:
+    def __init__(self, dir):
+        self.cur_id = 0
+        self.directory = dir
+        self.background_commands = {}
+        self.shell = DockerInteractive(id="default", workspace_dir=dir)
+
+    def run_command(self, command: str, background=False) -> CmdOutputObservation:
+        if background:
+            return self._run_background(command)
+        else:
+            return self._run_immediately(command)
+
+    def _run_immediately(self, command: str) -> CmdOutputObservation:
+        exit_code, output = self.shell.execute(command)
+        if exit_code != 0:
+            raise ValueError(
+                "Command failed with exit code " + str(exit_code) + ": " + output
+            )
```

Comment:
Let's remove this. Somewhere a try/catch got removed, and now failed commands cause the loop to die
```suggestion
```

Reply:
resolved!